zcash_client_sqlite/src/error.rs

```diff
@@ -185,12 +194,25 @@ impl fmt::Display for SqliteClientError {
             SqliteClientError::BalanceError(e) => write!(f, "Balance error: {}", e),
             SqliteClientError::NoteFilterInvalid(s) => write!(f, "Could not evaluate filter query: {:?}", s),
             #[cfg(feature = "transparent-inputs")]
-            SqliteClientError::ReachedGapLimit(account_id, bad_index) => write!(f,
-                "The proposal cannot be constructed until transactions with previously reserved ephemeral address outputs have been mined. \
-                 The ephemeral address in account {account_id:?} at index {bad_index} could not be safely reserved.",
+            SqliteClientError::ReachedGapLimit(key_scope, bad_index) => write!(f,
+                "The proposal cannot be constructed until a transaction with outputs to a previously reserved {} address has been mined. \
+                 The address at index {bad_index} could not be safely reserved.",
+                 match key_scope {
+                     KeyScope::Zip32(Scope::External) => "external transparent",
+                     KeyScope::Zip32(Scope::Internal) => "transparent change",
+                     KeyScope::Ephemeral => "ephemeral transparent",
```

Comment:
```suggestion
                     TransparentKeyScope::EXTERNAL => "external transparent",
                     TransparentKeyScope::INTERNAL => "transparent change",
                     TransparentKeyScope::EPHEMERAL => "ephemeral transparent",
                     _ => unreachable!(),
```

Reply:
Included in 0dcb2b2e74f6d8f0552c6cab01e2e0061150b3ce